### PR TITLE
Apply RegistryPublish in release job

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes
This PR applies sdkRegistryPublish in the release job, so it will generate PR in api-download automatically.

The generated test pr: https://github.com/mapbox/api-downloads/pull/1211

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -222,6 +222,12 @@ commands:
                   if [[ $CIRCLE_TAG == android-v* ||  $CIRCLE_BRANCH == main ]]; then
                     make sdkRegistryUpload
                   fi
+                  if [[ $CIRCLE_TAG == android-v* ]]; then
+                    git config --global user.email "MapboxCI@users.noreply.github.com"
+                    git config --global user.name "MapboxCI"
+                    export GITHUB_TOKEN=$(./mbx-ci github writer token)
+                    make sdkRegistryPublish
+                  fi
```
Comment (pengdev) [inline .circleci/config.yml:230]: @Chaoba could you add this reference CI integration to the documentation of sdk-regitstry plugin?

File: .circleci/config.yml [hunk 2]
```diff
@@ -222,6 +222,12 @@ commands:
                   if [[ $CIRCLE_TAG == android-v* ||  $CIRCLE_BRANCH == main ]]; then
                     make sdkRegistryUpload
                   fi
+                  if [[ $CIRCLE_TAG == android-v* ]]; then
+                    git config --global user.email "MapboxCI@users.noreply.github.com"
+                    git config --global user.name "MapboxCI"
+                    export GITHUB_TOKEN=$(./mbx-ci github writer token)
```
Comment (pengdev) [inline .circleci/config.yml:228]: nit: why use ./mbx-ci do we assume the CI is located at the same directory?

File: .circleci/config.yml (hunk 1 above)
Comment (Chaoba) [reply .circleci/config.yml:230]: Ok

File: .circleci/config.yml (hunk 2 above)
Comment (Chaoba) [reply .circleci/config.yml:228]: Yes, in `init-aws` command, we put it in the current directory.

File: .circleci/config.yml [hunk 3]
```diff
@@ -227,6 +227,12 @@ commands:
                   if [[ $CIRCLE_TAG == android-v* ||  $CIRCLE_BRANCH == main ]]; then
                     make sdkRegistryUpload
                   fi
+                  if [[ $CIRCLE_TAG == android-v* ]]; then
```
[the remark this answers is not in the document]
Comment (Chaoba) [reply .circleci/config.yml:230]: No, this is only needed during release. The upper `sdkRegistryUpload` is to upload the snapshot for main

**Review by pengdev — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]

**Pull request merged**